File: AprilTagScannerAndroid/app/src/main/kotlin/com/scanapriltag/models/TagLabels.kt
package com.scanapriltag.models

object TagLabels {
    fun shortFamily(family: String): String {
        val trimmed = family.trim()
        if (trimmed.isEmpty()) return ""
        return if (trimmed.startsWith("tag", ignoreCase = true)) trimmed.substring(3) else trimmed
    }

    fun format(family: String, id: Int, includeFamily: Boolean): String =
        if (includeFamily) "${shortFamily(family)}:$id" else id.toString()

    fun sessionNeedsFamily(records: List<TagRecord>, multiFamily: Boolean): Boolean =
        multiFamily || records.map { it.family }.distinct().size > 1
}

File: AprilTagScanner/Models/TagLabels.cs
namespace AprilTagScanner.Models;

public static class TagLabels
{
    public static string ShortFamily(string family)
    {
        if (string.IsNullOrWhiteSpace(family))
            return "";

        var trimmed = family.Trim();
        return trimmed.StartsWith("tag", StringComparison.OrdinalIgnoreCase)
            ? trimmed[3..]
            : trimmed;
    }

    public static string CanonicalFamily(string family)
    {
        var shortName = ShortFamily(family);
        return string.IsNullOrEmpty(shortName) ? "" : "tag" + shortName;
    }

    public static string Format(string family, int id, bool includeFamily) =>
        includeFamily ? $"{ShortFamily(family)}:{id}" : id.ToString();

    public static bool SessionNeedsFamily(IEnumerable<TagRecord> records, bool multiFamily) =>
        multiFamily || records.Select(r => r.Family).Distinct().Skip(1).Any();
}
